### PR TITLE
Build openssl from sources

### PR DESCRIPTION
**Pull request by gthvn1**

### CODE REVIEW

**Comment by gthvn1** [comment]
http://koji.xcp-ng.org/taskinfo?taskID=56757

File: SPECS/openssl.spec [hunk 1]
```diff
@@ -0,0 +1,2049 @@
...
+%{_sysconfdir}/pki/tls/misc/*.pl
+%{_sysconfdir}/pki/tls/misc/tsget
+
+%post libs -p /sbin/ldconfig
+
+%postun libs -p /sbin/ldconfig
+
+%changelog
+* Fri Jun 14 2024 Guillaume Thouvenin <guillaume.thouvenin@vates.tech> - 1:1.0.2k-26.1
+- Fixes RSA key verify error
```
Comment (stormi) [inline SPECS/openssl.spec]: It's not clear what this RSA key verify error is unless you go read the github link.
```suggestion
- Add openssl-1.0.2k-update-expiring-certificates.patch to fix build time tests
```

**Review by stormi — COMMENTED** [review]
IIRC the reason why we chose to rebuild openssh 1.0.2 instead of getting it directly from CentOS 7 was so that we could disable weak ciphers. Shouldn't this be done in this same PR so that we just do one build?

**Comment by gthvn1** [comment]
> IIRC the reason why we chose to rebuild openssh 1.0.2 instead of getting it directly from CentOS 7 was so that we could disable weak ciphers. Shouldn't this be done in this same PR so that we just do one build?

What do you mean by getting it directly? import it? Because without my fixes the import won't build so we need to apply patches to have an updated version of openssl.

**Comment by gthvn1** [comment]
But I agree that if some patches are required for weak cipher we can do it in the same build. @bleader you can add a commit on this PR or I can add it myself with your help. Or another PR as you prefer :)

**Review by stormi — APPROVED** [review]
After discussion, we established that this PR is ok as it is.

**Review by bleader — APPROVED** [review]
We checked, the build of this version disables weak ciphers already, no need for additional patch on this side.

**Comment by stormi** [comment]
Koji build to v8.3-incoming: http://koji.xcp-ng.org/taskinfo?taskID=56913

**Pull request merged**